.github/workflows/docs.yaml

```diff
@@ -0,0 +1,140 @@
+          --no-daemon --console plain \
+          -PsourceMaven=/home/runner/.m2/repository \
+          -PcentralMaven=https://repo.maven.apache.org/maven2/ \
+          -PtargetMaven=${{ github.workspace }}/docs/repo 2>&1 | tee build-docs.log      
+
+      - name: Upload build log
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: docs-gradle-build-log
+          path: docs/build-docs.log
+          retention-days: 7
+      
+      - name: Upload docs artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: docs-maven
+          path: docs/repo
+
+      - name: Upload docs site
+        uses: actions/upload-artifact@v4
+        with:
+          name: galasa-docs-site
+          path: docs/build/site
+
+  deploy-docs-pr-preview-to-github-pages:
+
+    if: ${{ (github.ref_name == 'main') }}
+    permissions:
+      contents: write
+
+    # Add a dependency to the build job
+    needs: build-docs
+
+    # Specify runner + deployment step
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout galasa-docs-preview repository
+        uses: actions/checkout@v4  
+        with:
+          repository: galasa-dev/galasa-docs-preview
+          path: galasa-docs-preview
+
+      - name: Clean out the docs which are currently in the main branch of the preview repo.
+        run: |-           
+          rm -fr galasa-docs-preview/docs
+
+      - name: Download the docs zip we built
+        id: download--built-docs
+        continue-on-error: false
+        uses: actions/download-artifact@v4
+        with:
+          name: galasa-docs-site
+          path: galasa-docs-preview/docs
+
+      - name: Push the changes to the preview git repo.
+        env: 
+          GITHUB_PAT: ${{ secrets.GALASA_TEAM_GITHUB_TOKEN_AUTH_IN_WORKFLOWS }}
+        run: |- 
+          chmod +x galasa-dev/galasa-docs-preview.git
+          cd galasa-docs-preview
+          git config --global user.name galasa-team
+          git config --global user.email galasa.team@gmail.com
+          git config --global credential.username galasa-team
+          git config user.email
+          git config user.name
+          # git remote set-url origin https://x-access-token:${{ env.GITHUB_PAT }}@github.com/galasa-dev/galasa-docs-preview
+          git add .
+          git commit -m "Automated update from the galasa-dev/galasa repo." \
+                     -m "Source branch ${{ env.BRANCH }} git head commit id: ${{ github.event.head_commit.id }}" \
+                     -m "Change: ${{ github.event.head_commit.message }}"
+          git push https://galasa-team:${{ env.GITHUB_PAT }}@github.com/galasa-dev/galasa-docs-preview.git main
+
+  report-failure:
+    # Skip this job for forks
+    if: ${{ failure() && github.repository_owner == 'galasa-dev' }}
+    name: Report failure in workflow
+    runs-on: ubuntu-latest
+    needs: [build-docs, deploy-docs-pr-preview-to-github-pages]
```

Comment:
Minor but `build-docs` can be removed from this array, as `deploy-docs-pr-preview-to-github-pages` couldn't run without that being complete anyway.

Reply:
Done